### ISSUE TITLE
Backoff not sufficient

### ISSUE THREAD

**Issue by peterbe** [mozilla-services/buildhub#425]
See https://sentry.prod.mozaws.net/operations/buildhub-stage/issues/1594248/

I wrote in my comment:

> https://archive.mozilla.org/pub/firefox/nightly/2018/04/2018-04-24-01-36-04-mozilla-central-l10n/ does indeed exist. Now. That means we didn't back off long enough. 

This is a lambda event that depends on fetching https://archive.mozilla.org/pub/firefox/nightly/2018/04/2018-04-24-01-36-04-mozilla-central-l10n/ and it tried 3 times and eventually had to give up. Since the URL now 200 OKs, it means it didn't wait long enough. 

We recently made it so that the `latest-inventory-to-kinto` doesn't do backoff in `fetch_json`, that it only does that in the lambda function. We can now increase the backoff configuration to either try more times or using longer pauses.

**Comment by peterbe** [comment]
@leplatrem Can you sanity check this issue? It seems we could simply increase the backoff times and seconds to a higher number(s) since it only matters for the lambda function. 
We have a cap of 5 minutes (I think) but it should be sufficiently long to retry more patiently.

**Comment by peterbe** [comment]
Here's another example. At the time of writing this, https://archive.mozilla.org/pub/firefox/nightly/2018/04/2018-04-25-10-01-22-mozilla-central/firefox-61.0a1.en-US.linux-i686.json is a perfectly fine 200 OK. But in https://sentry.prod.mozaws.net/operations/buildhub-stage/issues/4284392/ it failed with a ClientResponseError

**Comment by leplatrem** [comment]
Indeed, we could retry more times, or some to some exponential internal (https://github.com/litl/backoff/blob/master/backoff/_wait_gen.py)

5min will give us a pretty good margin. However, we have to be prepared for the fact that sometimes the `*-l10n/` folder may take more time to appear.
The code was supposed handle it like this:
https://github.com/mozilla-services/buildhub/blob/51d9563a1898fdd29a5aa782b6bcfd5c63682e58/jobs/buildhub/lambda_s3_event.py#L223-L224

Which is different from the case in https://sentry.prod.mozaws.net/operations/buildhub-stage/issues/4284392/ where we can see in the logs:
```
- Processing firefox nightly metadata: pub/firefox/nightly/2018/04/2018-04-26-10-00-55-mozilla-central/firefox-61.0a1.en-US.win32.json
- Fetch new nightly metadata
- GET 'https://archive.mozilla.org/pub/firefox/nightly/2018/04/2018-04-26-10-00-55-mozilla-central/firefox-61.0a1.en-US.win32.json'
- Backing off fetch_json(...) for 0.4s (aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found')
```

Honestly in this case I find it very weird that it takes so much time from the S3 event to the apparition on JSON API via the [bucket lister](https://github.com/mozilla-services/product-delivery-tools/tree/master/bucketlister). Maybe oremj has ideas...

Otherwise, using an AWS client we may have more chance to fetch it immediately...

**Comment by peterbe** [comment]
https://github.com/mozilla-services/buildhub/pull/432 just sets it to 6. I'll check with Wei that this isn't overwritten in the env of Stage or Prod. 

I'm not excited to dwell on this much more. 48 seconds is well less than 5 minutes (AWS Lambda max) and God forbid it takes longer than 48 seconds, the scraper will have to fix it later. Also, when we get the new `buildhub.json` we don't have to read any JSON listings at all any more anyway.

**Comment by peterbe** [comment]
It happened again :(
Even though we have `retry_on_notfound=True` and plenty of backoff it happened. 
https://sentry.prod.mozaws.net/operations/buildhub-stage/issues/4326570/
I'm just going to increase the backoff a bit more.

**Comment by peterbe** [comment]
Oops. In https://github.com/mozilla-services/buildhub/pull/466 I forgot to use the "fixes #425" suffix.
